node_package/src/createReactOutput.ts

```diff
@@ -45,13 +45,15 @@ export default function createReactOutput({
       // We just return at this point, because calling function knows how to handle this case and
       // we can't call React.createElement with this type of Object.
       return (renderFunctionResult as ServerRenderResult);
-    } // else we'll be calling React.createElement
-    // TODO: Can we detect if this is a React Element or a React Function Component?
-    // If already a ReactElement, then just return it.
-    // If a component, then wrap in an element
+    }
 
-    const reactComponent = renderFunctionResult as ReactComponent;
+    if (React.isValidElement(renderFunctionResult)) {
+      // If already a ReactElement, then just return it.
+      return renderFunctionResult;
+    }
```

Comment:
Basically, this skips the maybe essential call of React.createElement(component, props).

This 100% breaks if hooks are used.

Should we just error out and force the change to wrap the result JSX in a function with zero params? 

or should we make the upgrade easier for 99% of the cases that did not use Hooks.